### PR TITLE
Refactor visibility stores and managers

### PR DESCRIPTION
**Pull request by alexshtin**
**What changed?**
- Consolidate all visibility managers into a single one.
- Remove visibility sampling.
- Remove unused visibility QPS settings/ratelimiters.
- Add visibility persistence rate limiters.
- Move visibility persistence tests to `visibility` package.
- Restructure visibility sub packages.

**Why?**
Cleanup visibility code, remove unused code.

**How did you test it?**
Modified existing tests, local runs.

**Potential risks**
Minimum. Even PR is large it is mostly about reorg files.

**Is hotfix candidate?**
No.

### CODE REVIEW

File: common/persistence/visibility/factory.go [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+	esProcessorConfig *elasticsearch.ProcessorConfig,
+	searchAttributesProvider searchattribute.Provider,
+	searchAttributesMapper searchattribute.Mapper,
+
+	standardVisibilityPersistenceMaxReadQPS dynamicconfig.IntPropertyFn,
+	standardVisibilityPersistenceMaxWriteQPS dynamicconfig.IntPropertyFn,
+	advancedVisibilityPersistenceMaxReadQPS dynamicconfig.IntPropertyFn,
+	advancedVisibilityPersistenceMaxWriteQPS dynamicconfig.IntPropertyFn,
+	enableAdvancedVisibilityRead dynamicconfig.BoolPropertyFnWithNamespaceFilter,
+	advancedVisibilityWritingMode dynamicconfig.StringPropertyFn,
```
Comment (wxing1292) [inline common/persistence/visibility/factory.go:57]: consider wrap this within a struct? e.g. VisibilityOptions

File: common/persistence/visibility/manager/visibility_manager.go [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+	commonpb "go.temporal.io/api/common/v1"
+	enumspb "go.temporal.io/api/enums/v1"
+	workflowpb "go.temporal.io/api/workflow/v1"
+
+	"go.temporal.io/server/common/persistence"
+)
+
+type (
+	// VisibilityManager is used to manage the visibility store
+	VisibilityManager interface {
```
Comment (wxing1292) [inline common/persistence/visibility/manager/visibility_manager.go:42]: this file is copy pasted?

**Review by wxing1292 — APPROVED** [review]
mostly moving things around & adding necessary throttling & wrapper?

File: common/persistence/visibility/manager/visibility_manager.go (hunk 2 above)
Comment (alexshtin) [reply common/persistence/visibility/manager/visibility_manager.go:42]: yes.

File: common/persistence/visibility/factory.go (hunk 1 above)
Comment (alexshtin) [reply common/persistence/visibility/factory.go:57]: I was thinking about it but I also has an idea of splitting this manager into two: read and write. History service does only writes and frontend only reads. And then these 6 parameters will also split into 3+3.
Btw, should I just inject dynamic config client instead and read whatever I need from the manger itself?

**Comment by alexshtin** [comment]
Yes. mostly removing unnecessary wrappers. There were ~5 different visibility managers before and now there are only 2.

**Pull request merged**